### PR TITLE
Dev 1416 backups

### PR DESCRIPTION
**Pull request by AntonZelenin**

### CODE REVIEW

File: agent/Dockerfile [hunk 1]
```diff
@@ -20,4 +20,10 @@ ENV LOG_FILE_PATH="/var/log/agent.log" ANODOT_API_URL="http://api.anodot.com" EN
 ENV UWSGI_INI="/usr/src/app/uwsgi.ini"
 
 RUN pip install --upgrade pip && pip install -r requirements.txt && python setup.py install && echo ". /usr/src/app/agent-complete.sh" >> /root/.bashrc
-RUN apt update && apt install -y vim netcat curl nano
+
```
Comment (dariakharlan) [inline agent/Dockerfile:23]: Let's create a shell script, too many RUN commands and it's duplicated for the two images

File: agent/src/agent/cli/backup.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+@click.command()
+def backup():
+    filename = os.path.join(BACKUP_DIRECTORY, f'{AGENT_DB}_{datetime.now():%Y-%m-%d_%H:%M:%S}.dump')
+    if os.system(f'pg_dump -Fc -h {AGENT_DB_HOST} -U {AGENT_DB_USER} {AGENT_DB} > {filename}') == 0:
+        click.secho(f'{AGENT_DB} database successfully dumped to {filename}', fg='green')
+
+
+@click.command()
+@click.argument('dump_file')
```
Comment (dariakharlan) [inline agent/src/agent/cli/backup.py:16]: https://click.palletsprojects.com/en/7.x/arguments/#file-arguments

File: agent/src/agent/cli/backup.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+def restore(dump_file):
+    if not os.path.isfile(dump_file):
+        click.secho(f'Cannot open {dump_file}, make sure you enter the correct path', err=True, fg='red')
+        return
+    if click.confirm(f'Are you sure you want to restore `{AGENT_DB}` database from the dump? All current data in the database will be overwritten'):
+        if os.system(f'pg_restore -c -h {AGENT_DB_HOST} -U {AGENT_DB_USER} -d {AGENT_DB} {dump_file}') == 0:
```
Comment (dariakharlan) [inline agent/src/agent/cli/backup.py:22]: what about streamsets?

File: scripts/docker-compose.yml [hunk 4]
```diff
@@ -38,6 +39,8 @@ services:
       POSTGRES_PASSWORD: agent
       POSTGRES_DB: agent
     volumes:
+#      todo what path?
+      - backup-data:/backup-data
```
Comment (dariakharlan) [inline scripts/docker-compose.yml:43]: `- backup-data:/usr/src/app/backup-data`

**Review by dariakharlan — CHANGES_REQUESTED** [review]

File: agent/src/agent/cli/backup.py (hunk 2 above)
Comment (AntonZelenin) [reply agent/src/agent/cli/backup.py:16]: I'll get file handle if I use it, but I need only a path to a file

**Review by dariakharlan — APPROVED** [review]

**Pull request merged**